src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2599,16 +2598,11 @@ public void LowMemoryOver()
 
         internal QueryBuilderFactories GetQueryBuilderFactories()
         {
-            //We don't want to always generate this object but if we generate it mutiple times at startup it isn't really importent.
-            if (_queryBuilderFactories == null)
+            return new QueryBuilderFactories
```

Comment:
Need to allocate this once, there is nothing that is unique per call

Reply:
that was the original implementation, you commented that i shouldn't do it so

Reply:
I meant that we want to allocate it once, without the if, because there is nothing special about the call that require a separate instance.

Reply:
For that matter, let's change `QueryBuilderFactories` to call directly to the relevant methods, instead of using delegate indirection